src/test/signal/test_signals.rs

```diff
@@ -888,6 +889,61 @@ fn test_sigaltstack_autodisarm() -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+static GLOBAL_STATIC: u32 = 0xdeadbeef;
+extern "C" fn change_rax_from_null_to_global_static(
+    signal: i32,
+    info: *mut libc::siginfo_t,
+    voidctx: *mut std::ffi::c_void,
+) {
+    assert_eq!(signal, signal::SIGSEGV as i32);
+
+    let info = unsafe { info.as_mut().unwrap() };
+    // Not exposed in libc crate
+    const SEGV_MAPERR: i32 = 1;
+    assert_eq!(info.si_code, SEGV_MAPERR);
+    assert!(unsafe { info.si_addr().is_null() });
+
+    let ctx = voidctx as *mut libc::ucontext_t;
+    let ctx = unsafe { ctx.as_mut().unwrap() };
+    ctx.uc_mcontext.gregs[libc::REG_RAX as usize] = &GLOBAL_STATIC as *const u32 as i64;
+}
+
+fn test_synchronous_sigsegv() -> Result<(), Box<dyn Error>> {
+    // Ensure SIGSEGV isn't blocked.
+    let mut sigset = signal::SigSet::empty();
+    sigset.add(signal::SIGSEGV);
+    signal::sigprocmask(signal::SigmaskHow::SIG_UNBLOCK, Some(&sigset), None).unwrap();
+
+    // Install our SIGSEGV handler, which is going to mutate registers in the
+    // caller to fix the null pointer dereference below.
+    // This may seem esoteric, but this is sometimes done to implement custom
+    // memory management. Observed in OpenJDK in particular - see
+    // https://github.com/shadow/shadow/issues/2094.
+    unsafe {
+        signal::sigaction(
+            signal::SIGSEGV,
+            &signal::SigAction::new(
```

Comment:
Unregister this handler at the end of the test to prevent it from causing weird behaviour in later tests?

Reply:
Good point; done